dipy/reconst/tests/test_csdeconv.py

```diff
@@ -7,19 +7,96 @@
 from dipy.sims.voxel import (multi_tensor,
                              single_tensor,
                              multi_tensor_odf,
-                             all_tensor_evecs)
+                             all_tensor_evecs, single_tensor_odf)
 from dipy.core.gradients import gradient_table
 from dipy.core.sphere import HemiSphere
 from dipy.reconst.csdeconv import (ConstrainedSphericalDeconvModel,
                                    ConstrainedSDTModel,
                                    forward_sdeconv_mat,
                                    odf_sh_to_sharp,
-                                   auto_response,
+                                   auto_response, recursive_response,
                                    csd_predict)
-from dipy.reconst.peaks import peak_directions, default_sphere
+from dipy.reconst.peaks import (peak_directions, default_sphere)
 from dipy.core.sphere_stats import angular_similarity
 from dipy.reconst.shm import (sf_to_sh, sh_to_sf, QballModel,
-                              CsaOdfModel, sph_harm_ind_list)
+                              CsaOdfModel, sph_harm_ind_list,
+                              sph_harm_ind_list, real_sph_harm)
+from dipy.reconst.shm import lazy_index
+from dipy.core.geometry import cart2sphere
+import dipy.reconst.dti as dti
+from dipy.reconst.dti import fractional_anisotropy
+from dipy.core.sphere import Sphere
+
+
+def test_recursive_response_calibration():
+    SNR = 100
```

Comment:
@MrBago, yes you are right, I noticed a failure sometimes as well. I think it has to do with the 'fat' response function that I choose for the first iteration, this sometimes doesn't converge to the right solution. I am not sure why this is, in the simulations with our own CSD implementation this never  seems to happen. So either I have to increase the anisotorpy of the initial RF for this test and check if it improves, or we have to look deeper in the code (in real data I did not see this issue).